src/Agent.Worker/Release/ReleaseCommandExtension.cs

```diff
@@ -12,76 +12,68 @@
 
 namespace Microsoft.VisualStudio.Services.Agent.Worker.Release
 {
-    public sealed class ReleaseCommandExtension : AgentService, IWorkerCommandExtension
+    public sealed class ReleaseCommandExtension : BaseWorkerCommandExtension
     {
-        public Type ExtensionType => typeof(IWorkerCommandExtension);
-
-        public string CommandArea => "release";
-
-        public HostTypes SupportedHostTypes => HostTypes.Release | HostTypes.Deployment;
-
-        public void ProcessCommand(IExecutionContext context, Command command)
+        public ReleaseCommandExtension()
         {
-            if (string.Equals(command.Event, WellKnownReleaseCommand.UpdateReleaseName, StringComparison.OrdinalIgnoreCase))
-            {
-                ProcessReleaseUpdateReleaseNameCommand(context, command.Data);
-            }
-            else
-            {
-                throw new Exception(StringUtil.Loc("RMCommandNotFound", command.Event));
-            }
+            CommandArea = "release";
+            SupportedHostTypes = HostTypes.Release | HostTypes.Deployment;
+            InstallWorkerCommand(new ReleaseUpdateReleaseNameCommand());
         }
 
-        private void ProcessReleaseUpdateReleaseNameCommand(IExecutionContext context, string data)
+        private class ReleaseUpdateReleaseNameCommand: IWorkerCommand
         {
-            ArgUtil.NotNull(context, nameof(context));
-            ArgUtil.NotNull(context.Endpoints, nameof(context.Endpoints));
-
-            Guid projectId = context.Variables.System_TeamProjectId ?? Guid.Empty;
-            ArgUtil.NotEmpty(projectId, nameof(projectId));
-            
-            string releaseId = context.Variables.Release_ReleaseId;
-            ArgUtil.NotNull(releaseId, nameof(releaseId));
+            public string Name => "updatereleasename";
+            public string[] Aliases => new string [] {};
```

Comment:
are array mutable in C#? I do not think so. 
If so, perhaps `null` is a better initializer here.

Reply:
changed to List<string> and initialized to null in this case